questionnaires/templates/quizzes/quiz.html

```diff
@@ -1,21 +1,55 @@
 {% extends "base.html" %}
-{% load wagtailcore_tags %}
-{% load i18n %}
+{% load static i18n questionnaires_tags wagtailcore_tags %}
+
+{% block extra_css %}
+    <link rel="stylesheet" type="text/css" href="{% static 'css/quiz.css' %}">
+{% endblock %}
 
 {% block content %}
-    <h1>{{ page.title }}</h1>
-    {{ form.non_field_errors }}
-    <div>
-        <p>{{ page.description }}</p>
-        <hr>
-        <form action="{% pageurl page %}{% if self.multi_step or self.has_page_breaks %}?p={{ fields_step.number|add:"1" }}{% endif %}" method="POST">
-            {% csrf_token %}
-            <div>
-                {{ form.as_p }}
-            </div>
-            <div>
-                <input type="submit" value="{% trans page.submit_button_text %}"/>
-            </div>
-        </form>
+    {% include 'back_button.html' with back_url=back_url %}
+    <div class="quiz-page">
+        <div class="container">
+            <h1 class="title quiz-page__title">{% trans page.title %}</h1>
+            <form action="
+                    {% pageurl page %}{% if self.multi_step or self.has_page_breaks %}?p={{ fields_step.number|add:"1" }}&back_url=
+                        {{ request.path }}{% else %}?back_url={{ request.path }}{% endif %}"
+                  method="POST" class="quiz-page__content">
+                {% csrf_token %}
+                {% for field in form %}
+                    <div class="quest-item">
+                        <div class="quest-item__header">
+                            <p class="quest-item__number">{{ forloop.counter }}
+                                of
+                                {% if self.multi_step or self.has_page_breaks %}
+                                    {{ fields_step.paginator.count }}{% else %}
+                                    {{ form.fields | length }}{% endif %}
+                                questions</p>
+                            <p class="quest-item__step-desc">
+                                <span>{% trans "Check all that apply" %}</span>
+                            </p>
+                        </div>
+                        <div class="quest-item__desc">
+                            <p>{% trans field.label %}</p>
+                        </div>
+                        <div class="quest-item__content" tabindex="0">
+                            {% render_checkboxes field %}
```

Comment:
This rendering, too, should depend on the input type. See e.g. questionnaires/templates/survey/survey.html